File: LeetCode75-sols/Array_String/CPP/String_compression.cpp
class Solution {
public:
    int compress(vector<char>& chars) {
        if (size(chars) == 1) {
            return 1;
        }
        int j = 0;
        int count = 1;
        for (int i = 0; i < size(chars); i++) {
            count = 1;
            while ((i < (size(chars) - 1)) && (chars[i] == chars[i+1])) {
                count += 1;
                i += 1;
            }

            chars[j] = chars[i];
            j += 1;

            if (count > 1) {
                for (int k = 0; k < to_string(count).length();  k++) {
                    chars[j] = to_string(count)[k];
                    j += 1;
                } 
            }
        }
        return j;
    }
};

File: LeetCode75-sols/Array_String/Python/String_compression.py
class Solution:
    def compress(self, chars: List[str]) -> int:
        n = len(chars)
        if n==1:
            return 1
        # i is left pointer, j is right ptr
        i,j = 0,0
        while i<n:
            count = 1
            while ((i < (n-1)) and (chars[i] == chars[i+1])):
                count += 1
                i += 1
            
            # Set left most entry in final chars array then increment j,i
            chars[j] = chars[i]
            j += 1
            i += 1

            # Input length of compressed characters in chars as string
            if count > 1:
                for k in range(len(str(count))):
                    chars[j] = str(count)[k]
                    j += 1

        return j